lib/sycamore/sycamore/tests/unit/test_docset.py

```diff
@@ -199,3 +199,22 @@ def test_map_elements(self):
         for doc in all_docs:
             for elem in doc.elements:
                 assert elem.properties["element_double"] == elem.properties["element_val"] * 2
+
+    def test_filter_elements(self):
+        docs = []
+        for i in range(10):
+            doc = Document(text_representation=f"Document {i}", doc_id=i, properties={"document_number": i})
+            doc.elements = [
+                Element(text_representation=f"Document {i} Element {j}", properties={"element_val": j})
+                for j in range(10)
+            ]
+
+        context = sycamore.init()
+        docset = context.read.document(docs).filter_elements(lambda e: e.properties["element_val"] % 2 == 0)
+
+        all_docs = docset.take_all()
+        assert len(all_docs) == len(docs) / 2
```

Comment:
wait there was no explode. shouldn't these be the same length? (but each doc only has 5 elts?)

Reply:
oh it's bc you never appended the constructed `doc`s to `docs` so 0 = 0/2

Reply:
Oh oops. This means the test I copied this from was wrong too.